ydb/docs/en/core/yql/reference/yql-core/udf/list/knn.md

```diff
@@ -0,0 +1,100 @@
+## Introduction
+
+[Nearest Neighbor search](https://en.wikipedia.org/wiki/Nearest_neighbor_search) (NN) is an optimization task that consists in finding the closest point (or set of points) in a given dataset to a given query point. Closeness can be defined in terms of distance or similarity metrics.
+A generalization of the NN problem is the k-NN problem, where it's required to find the k nearest points to the query point. This can be useful in various applications such as image classification, recommendation systems, and more.
+https://en.wikipedia.org/wiki/Nearest_neighbor_search
```

Comment:
```suggestion
[Nearest Neighbor search](https://en.wikipedia.org/wiki/Nearest_neighbor_search) (NN) is an optimization task that consists of finding the closest point in a given dataset to a given query point. Closeness can be defined in terms of distance or similarity metrics.
A generalization of the NN problem is the [k-NN problem](https://en.wikipedia.org/wiki/K-nearest_neighbors_algorithm), where it's required to find the `k` nearest points to the query point. This can be useful in various applications such as image classification, recommendation systems, etc.
https://en.wikipedia.org/wiki/Nearest_neighbor_search
```